src/main/java/org/broadinstitute/hellbender/engine/spark/SparkCommandLineArgumentCollection.java

```diff
@@ -20,6 +20,9 @@
     @Argument(fullName = "sparkMaster", doc="URL of the Spark Master to submit jobs to when using the Spark pipeline runner.", optional = true)
     private String sparkMaster = SparkContextFactory.DEFAULT_SPARK_MASTER;
 
+    @Argument(fullName = "sparkDeployMode", doc="Mode of the spark driver. Default value: client. Possible values: {client, cluster}.", optional = true)
+    private String sparkDeployMode = "client";
```

Comment:
I guess that this should be set in the `SparkContext` for the `SparkCommandLineProgram` using the `SparkContextFactory`. But as a suggestion for downstream projects: can it be possible to make `SparkCommandLineArgumentCollection` abstract (or an interface), and pass the specified sparkMaster to the  map in `getSparkProperties()` (I guess that using `propertyMap.put("spark.master", sparkMaster)` should work as expected...

Reply:
As the deploy-mode is related to the spark driver, a.k.a. sparkContext, I put it with the sparkMaster instead of more jobs-related spark parameters (as the programName). However, I could and I should do an accessor (getSparkDeployMode) and try to use it.
At first, I checked if SparkContext was able to give the default deploy-mode (which is 'client' most of the time) to do something as `private String sparkDeployMode = SparkContextFactory.DEFAULT_SPARK_DEPLOYMODE;` but I failed. I could invest more time and add it myself but I didn't get enough.

I don't think I really undestand your second part. Do you mean `getSparkProperties` should include the sparkMaster property too ? Is it not already the case ? (I didn't check this part :/).

Note: I'm not a GATK developer. I'm just someone who need the ability to set the deploy-mode. As I had to edit the code to remove a "bug" (the UserClassPathFirst property set to true instead of the default false), I took the opportunity to implement the deploy-mode parameter to avoid to do `--conf spark.submit.deployMode=cluster`. I'm not sure I have done it in the GATK way...